python/cuml/dask/common/comms_base.py

```diff
@@ -0,0 +1,413 @@
+        """
+        return list(self.client.has_what().keys())
+
+    def worker_ranks(self):
+        """
+        Builds a dictionary of { (worker_address, worker_port) : worker_rank }
+        """
+        return dict(list(map(lambda x: (x[1], x[0]), self.nccl_clique)))
+
+    def worker_ports(self):
+        """
+        Builds a dictionary of { (worker_address, worker_port) : worker_port }
+        """
+        return dict(list(self.ucp_ports))
+
+    def worker_info(self):
+        """
+        Builds a dictionary of { (worker_address, worker_port) :
+                                (worker_rank, worker_port ) }
+        """
+        ranks = self.worker_ranks()
+        ports = self.worker_ports() if self.comms_p2p else None
+
+        if self.comms_p2p:
+            output = {}
+            for k in self.worker_ranks().keys():
+                output[k] = (ranks[k], ports[k])
+            return output
+
+        else:
+            return ranks
+
+    @staticmethod
+    def func_init_nccl(workerId, nWorkers, uniqueId):
+        """
+        Initialize ncclComm_t on worker
+        :param workerId: int ID of the current worker running the function
+        :param nWorkers: int Number of workers in the cluster
+        :param uniqueId: array[byte] The NCCL unique Id generated from the
+                         client.
+        """
+        n = nccl()
+        n.init(nWorkers, uniqueId, workerId)
+        return n
+
+    @staticmethod
+    def func_get_ucp_port(sessionId, r):
+        """
+        Return the port assigned to a UCP listener on worker
+        :param sessionId: uuid Unique session id for current instance
+        :param r: float a random number to stop the function from being cached
+        """
+        dask_worker = get_worker()
+        port = dask_worker.data[sessionId].port
+        return port
+
+    @staticmethod
+    async def ucp_create_listener(sessionId, r):
+        """
+        Creates a UCP listener for incoming endpoint connections.
+        This function runs in a loop asynchronously in the background
+        on the worker
+        :param sessionId: uuid Unique id for current instance
+        :param r: float a random number to stop the function from being cached
+        """
+
+        dask_worker = get_worker()
+        if sessionId in dask_worker.data:
+            print("Listener already started for sessionId=" +
+                  str(sessionId))
+        else:
+            ucp.init()
+            listener = ucp.start_listener(connection_func, 0,
+                                          is_coroutine=True)
+
+            dask_worker.data[sessionId] = listener
+            task = asyncio.create_task(listener.coroutine)
+
+            while not task.done():
+                await task
+                await asyncio.sleep(1)
+
+            ucp.fin()
+            del dask_worker.data[sessionId]
+
+    @staticmethod
+    def ucp_stop_listener(sessionId, r):
```

Comment:
```suggestion
    def func_ucp_stop_listener(sessionId, r):
```

I could see an argument that function names get too long with this naming convention